AVFoundation-Combine/Publishers/KVObservingPublisher.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  KVObservingPublisher.swift
+//  AVFoundation-Combine
+//
+//  Created by József Vesza on 2020. 07. 17..
+//  Copyright © 2020. József Vesza. All rights reserved.
+//
+
+import Foundation
+import Combine
+import AVKit
+
+public extension Publishers {
+    struct KVObservingPublisher<ObservedType: NSObject, ObservedValue>: Publisher {
+        public typealias Failure = Never
+        public typealias Output = ObservedValue
+        
+        private let observedObject: ObservedType
+        private let keyPath: KeyPath<ObservedType, ObservedValue>
+        
+        init(observedObject: ObservedType, keyPath: KeyPath<ObservedType, ObservedValue>) {
+            self.observedObject = observedObject
+            self.keyPath = keyPath
+        }
+        
+        public func receive<S>(subscriber: S) where S : Subscriber, Self.Failure == S.Failure, Self.Output == S.Input {
+            let subscription = KVObservingSubscription(subscriber: subscriber,
+                                                       observedObject: observedObject,
+                                                       keyPath: keyPath)
+            subscriber.receive(subscription: subscription)
+        }
+    }
+    
+    private final class KVObservingSubscription<S: Subscriber, ObservedType: NSObject, ObservedValue>: Subscription where S.Input == ObservedValue {
+        private var subscriber: S?
+        private var requested: Subscribers.Demand = .none
+        private var stateObserverToken: NSKeyValueObservation? = nil
```

Comment:
You might want to rename this to something more generic like `observationToken` or similar